File: CS/DXSample/SchedulingViewModel.cs
using System;
using System.Collections.ObjectModel;
using System.Data.Entity;
using System.Windows.Input;
using DevExpress.Mvvm;
using DXSample.Data;

namespace DXSample {
    public class SchedulingViewModel : ViewModelBase {        
        SchedulingContext context;
        public ObservableCollection<AppointmentEntity> Appts { get { return context.Appointments.Local; } }
        public ObservableCollection<ResourceEntity> Calendars { get { return context.Resources.Local; } }
        public string Status {
            get { return GetValue<string>(); }
            set { SetValue(value); }

        }               
        public SchedulingViewModel() {
            context = new SchedulingContext();
            context.Appointments.Load();
            context.Resources.Load();
        }
        public ICommand SaveCommand {
            get {
                return new DelegateCommand(() => {
                    context.SaveChanges();
                    Status = $"Changes are saved to database. {DateTime.Now.ToLongTimeString()}.";
                });
            }
        }
    }
}

File: VB/DXSample/ViewModels/SchedulingViewModel.vb
Imports System.Collections.ObjectModel
Imports System.Data.Entity
Imports System.Windows.Input
Imports DevExpress.Mvvm
Imports DXSample.Data

Namespace DXSample.ViewModels

    Public Class SchedulingViewModel
        Inherits ViewModelBase

        Private context As SchedulingContext

        Public ReadOnly Property Appts As ObservableCollection(Of AppointmentEntity)
            Get
                Return context.Appointments.Local
            End Get
        End Property

        Public ReadOnly Property Calendars As ObservableCollection(Of ResourceEntity)
            Get
                Return context.Resources.Local
            End Get
        End Property

        Public Property Status As String
            Get
                Return GetValue(Of String)()
            End Get

            Set(ByVal value As String)
                SetValue(value)
            End Set
        End Property

        Public Sub New()
            context = New SchedulingContext()
            context.Appointments.Load()
            context.Resources.Load()
        End Sub

        Public ReadOnly Property SaveCommand As ICommand
            Get
                Return New DelegateCommand(Sub()
                    context.SaveChanges()
                    Status = $"Changes are saved to database. {Date.Now.ToLongTimeString()}."
                End Sub)
            End Get
        End Property
    End Class
End Namespace
